beacon_node/http_api/src/publish_attestations.rs

```diff
@@ -0,0 +1,292 @@
+                            let reprocess_fn = move || {
+                                let result = verify_and_publish_attestation(
+                                    &reprocess_chain,
+                                    &attestation,
+                                    seen_timestamp,
+                                    &reprocess_network_tx,
+                                );
+                                // Ignore failure on the oneshot that reports the result. This
+                                // shouldn't happen unless some catastrophe befalls the waiting
+                                // thread which causes it to drop.
+                                let _ = tx.send(result);
+                            };
+                            let reprocess_msg =
+                                ReprocessQueueMessage::UnknownBlockUnaggregate(QueuedUnaggregate {
+                                    beacon_block_root,
+                                    process_fn: Box::new(reprocess_fn),
+                                });
+                            if reprocess_tx.try_send(reprocess_msg).is_err() {
+                                PublishAttestationResult::Failure(Error::ReprocessFull)
+                            } else {
+                                PublishAttestationResult::Reprocessing(rx)
+                            }
+                        }
+                        Err(Error::Validation(AttestationError::PriorAttestationKnown {
+                            ..
+                        })) => PublishAttestationResult::AlreadyKnown,
+                        Err(e) => PublishAttestationResult::Failure(e),
+                    }
+                })
+                .map(Some)
+                .collect::<Vec<_>>())
+        })
+        .await?;
+
+    // Asynchronously wait for re-processing of attestations to unknown blocks. This avoids blocking
+    // any of the beacon processor workers while we wait for reprocessing.
+    let (reprocess_indices, reprocess_futures): (Vec<_>, Vec<_>) = prelim_results
+        .iter_mut()
+        .enumerate()
+        .filter_map(|(i, opt_result)| {
+            if let Some(PublishAttestationResult::Reprocessing(..)) = &opt_result {
+                let PublishAttestationResult::Reprocessing(rx) = opt_result.take()? else {
+                    // Unreachable.
+                    return None;
+                };
+                Some((i, rx))
+            } else {
+                None
+            }
+        })
+        .unzip();
+    let reprocess_results = futures::future::join_all(reprocess_futures).await;
+
+    // Join everything back together and construct a response.
+    // This part should be quick so we just stay in the Tokio executor's async task.
+    for (i, reprocess_result) in reprocess_indices.into_iter().zip(reprocess_results) {
+        let Some(result_entry) = prelim_results.get_mut(i) else {
+            continue;
```

Comment:
This would be an unreachable scenario right? Is it worth logging an error here in case if we hit this line?

Reply:
Logging added in 6043b0cca1674ee3b78099d3c82c863d76944a10